pkg/apis/machine/validation/awsmachineclass.go

```diff
@@ -116,18 +116,34 @@ func validateAWSClassSpecTags(tags map[string]string, fldPath *field.Path) field
 
 func validateBlockDevices(blockDevices []machine.AWSBlockDeviceMappingSpec, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if len(blockDevices) > 1 {
-		allErrs = append(allErrs, field.Required(fldPath.Child(""), "Can only specify one (root) block device"))
-	} else if len(blockDevices) == 1 {
-		if blockDevices[0].Ebs.VolumeSize <= 0 {
-			allErrs = append(allErrs, field.Required(fldPath.Child("ebs.volumeSize"), "Please mention a valid ebs volume size"))
+
+	if 0 == len(blockDevices) {
+		allErrs = append(allErrs, field.Required(fldPath, "At least one disk is required"))
+	}
+	rootPartitionCount := 0
+	for i, disk := range blockDevices {
+		idxPath := fldPath.Index(i)
+
+		if disk.DeviceName == "/root" {
+			rootPartitionCount++
+		}
+
+		if disk.Ebs.VolumeSize <= 0 {
+			allErrs = append(allErrs, field.Required(idxPath.Child("ebs.volumeSize"), "Please mention a valid ebs volume size"))
 		}
 		if blockDevices[0].Ebs.VolumeType == "" {
-			allErrs = append(allErrs, field.Required(fldPath.Child("ebs.volumeType"), "Please mention a valid ebs volume type"))
+			allErrs = append(allErrs, field.Required(idxPath.Child("ebs.volumeType"), "Please mention a valid ebs volume type"))
 		} else if blockDevices[0].Ebs.VolumeType == "io1" && blockDevices[0].Ebs.Iops <= 0 {
-			allErrs = append(allErrs, field.Required(fldPath.Child("ebs.iops"), "Please mention a valid ebs volume iops"))
+			allErrs = append(allErrs, field.Required(idxPath.Child("ebs.iops"), "Please mention a valid ebs volume iops"))
 		}
 	}
+
+	if rootPartitionCount > 1 {
+		allErrs = append(allErrs, field.Required(fldPath, "Only one '/root' partition can be specified"))
+	} else if rootPartitionCount == 0 && len(blockDevices) > 1 {
```

Comment:
`len(blockDevices) > 1` allow backward compatibility when a single disk is provided without `DeviceName`

Reply:
Should this be added as comment in the code also